File: Assessment/src/pages/HomePage.py
class HomePage:

    def __init__(self, page):
        self.page = page
        self.lbl_title = page.locator("title")
        self.btn_home = page.locator("#nav-home > a")
        self.btn_contact = page.locator("#nav-contact > a")
        self.btn_shop = page.locator("#nav-shop > a")
        self.btn_cart = page.locator("#nav-cart > a")

    @property
    # function to get page title
    def get_title(self):
        return self.lbl_title

    # function to click home tab
    def click_home(self):
        self.btn_home.click()

    # function to click contact tab
    def click_contact(self):
        self.btn_contact.click()

    # function to click shop tab
    def click_shop(self):
        self.btn_shop.click()

    # function to click cart tab
    def click_cart(self):
        self.btn_cart.click()

File: Assessment/src/pages/HomePage.ts
import { Page } from '@playwright/test'

export default class HomePage { 
    constructor(private page: Page) {}

    lbl_title = 'title'
    btn_home = '//*[@id="nav-home"]/a'
    btn_contact = '//*[@id="nav-contact"]/a'
    btn_shop = '//*[@id="nav-shop"]/a'
    btn_cart = '//*[@id="nav-cart"]/a'

    
    // function to get page title
    async get_title() {
        return await this.page.locator(this.lbl_title).innerText()
    }

    // function to click home tab
    async click_home() {
        await this.page.click(this.btn_home)
    }

    // function to click contact tab
    async click_contact() {
        await this.page.click(this.btn_contact)
    }

    // function to click shop tab
    async click_shop() {
        await this.page.click(this.btn_shop)
    }

    // function to click cart tab
    async click_cart() {
        await this.page.click(this.btn_cart)
    }

}
